server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java

```diff
@@ -71,7 +72,8 @@
         new SwaggerRoutesModule(),
         new WebAdminServerModule(),
         new DLPRoutesModule(),
-        new SieveQuotaRoutesModule());
+        new SieveQuotaRoutesModule(),
+        new StatusRouteModule());
```

Comment:
the idea is really great but we should generalize that to a real healthcheck endpoint IMO

Reply:
What do you mean? `red/yellow/green` with various components defining their health and James doing a logical AND?

Reply:
There are plenty of way of doing that. Spring boot has a lot of built-in healthcheck features.
dropwizard has some too ( https://www.dropwizard.io/1.3.5/docs/manual/core.html#health-checks )
OBM has some : https://www.obm.org/wiki/obm-health-check-system
If you look at k8s documentation, any endpoint returning less than HTTP 400 on success is a good enough liveness probe.

IMO, having a simple health probe registry and a specific endpoint (no admin access required but maybe a specific port) to run the probes would be enough.